drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -379,10 +381,10 @@ NTSTATUS TCPConnect
         }
 
         /* Allocate the port in the port bitmap */
-        Connection->AddressFile->Port = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
-
-        /* This should never fail */
-        ASSERT(Connection->AddressFile->Port != 0xFFFF);
+        AllocatedPort = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
+        /* This should never fail unless all ports are in use */
+        ASSERT(AllocatedPort != (UINT)-1);
```

Comment:
Same comment as above, reporting correct NTSTATUS to caller instead of asserting seems more correct (exhausting all local ports is a scenario that can happen in a "normal" lifecycle)

Reply:
This was a request from @learn-more in the previous PR, but based on the assumption that "this should never fail" which was the code comment.
But you're right, the error case is correctly manage and this CAN happen under very loaded use cases (or poorly written networking app) so the best option is NOT to ASSERT but to fail gracefully (and i'd suggest to log an ERR since it should be very unlikely)

Reply:
Latest commit attempts to address this issue. Please review and approve if appropriate.